repo_policy_compliance/__init__.py

```diff
@@ -96,6 +97,40 @@ def _check_signed_commits_required(branch: Branch) -> Report:
     return Report(result=Result.PASS, reason=None)
 
 
+def _check_unique_commits_signed(
+    branch_name: str, other_branch_name: str, repository: Repository
+) -> Report:
+    """Check that the commits unique to a branch are signed.
+
+    Args:
+        branch_name: The name of the branch to check.
+        other_branch_name: The name of the branch which will be used to exclude commits.
+        repository: The repository the branches are on.
+
+    Returns:
+        Whether the unique commits on the branch are signed.
+    """
+    other_branch_commit_shas = {
+        commit.sha for commit in repository.get_commits(sha=other_branch_name)
+    }
+    branch_commits = repository.get_commits(sha=branch_name)
+    unique_branch_commits = (
+        commit for commit in branch_commits if commit.sha not in other_branch_commit_shas
+    )
```

Comment:
Curious comment/no need to change: Could this be replaced by unique = list(set(branch_commits)-set(other)) ? Would make a difference?

Reply:
The advantage of the current approach compared to the proposal is that the current approach uses a generator so it doesn't need to load all the commits into memory. Converting to a list means that all the commits are loaded. I believe that the GitHub client fetches the commits as required